arelle/plugin/validate/DBA/rules/__init__.py

```diff
@@ -52,6 +52,27 @@ def errorOnDateFactComparison(
         )
 
 
+def errorOnRequiredFact(
+        modelXbrl: ModelXbrl,
+        factQn: QName,
+        code: str,
+        message: str,
+) -> Iterable[Validation]:
+    """
+    Yields an error if a fact with the given QName is not tagged with a valid date and a non-nil value.
+    :return: Yields validation errors.
+    """
+    facts: set[ModelFact] = modelXbrl.factsByQname.get(factQn, set())
+    if facts:
+        for fact in facts:
+            if fact.xValue is not None:
```

Comment:
In the description above it states that a fact must be tagged `with a valid date and a non-nil value.` But you dont check that the facts date is acceptable in this code. Is that done elsewhere?

Reply:
That is a relic of a mis-reading of the spec.